### DISCUSSION TITLE
How can I change the playback speed of the demo and rewind at will?

### DISCUSSION THREAD

**Discussion by l0m-dev** [saul/demofile#255] [Questions] [answered]
How would this be done?

**Comment by master117** [comment]
What do you mean playback speed? This is a tool for analysis demos, it goes from the start to the end. What would playback speed achieve? Skip ticks?

Rewinding would practically mean analysing the demo from the beginning since many events build upon each other.

**Comment by l0m-dev** [comment]
I was thinking of saving the demo data after analyzing it so you can play it afterwards.

**Comment by master117** [comment, marked as answer]
You can not split/save parts of a demo (with this tool). If you save all of it you get the same file as before. You can not set the playback speed/behaviour of a demo file by editing the demo.

**Comment by saul** [comment]
I think @master117's comments answer your question. This library is for programmatically analysing a demo file - there is no notion of rewind or playback speed. The demo is parsed as quickly as your CPU can process it.

**Comment by FalkF** [comment]
Btw. wouldnt changing the playback speed be as simple as to double the tickrate that is probably somewhere in the demo header?

**Reply by master117** [reply]
Doubling the playback speed where? In CSGO? CSGO already has speed settings for playing back a demo. Within the analyzer? The analyzer already uses the maximum speed your CPU can utilize. 

The original question itself is just not matching what this library does. It's for going through all events/ticks of a demo as fast as possible and providing information on what happened in that tick.

It's not a demo viewer.
